UMN-VMWareRA.psm1

```diff
@@ -1197,6 +1197,321 @@ function Set-VMWareRAVMTagAttachedToObject {
+            }
+        }
+
+        ## Construct url
+        $url = "https://$vCenter/rest/vcenter/host"
+
+        [Array] $urlFilters = @();
+
+        if ($name)
+        {
+            $urlFilters += "filter.names=${name}";
+        }
+
+        if ($hostID)
+        {
+            $urlFilters += "filter.hosts=${hostID}";
+        }
+
+        if ($clusterID)
+        {
+            $urlFilters += "filter.clusters=${clusterID}";
+        }
+
+        if ($urlFilters.Count -gt 0)
+        {
+            $url += "?";
+
+            foreach ($urlFilterItem in $urlFilters)
+            {
+                $url += $urlFilterItem + "&";
+            }
+        }
+
+        $return = Invoke-WebRequest -Uri $url -Method Get -Headers @{'vmware-api-session-id'=$sessionID} -ContentType 'application/json' -UseBasicParsing
+        return(($return.Content | ConvertFrom-Json).value)
+    }
+    End
+    {
+    }
+}
+#endregion
+
+#region Get-VMWareRAVMByName
+function Get-VMWareRAVMByName {
+<#
+    .Synopsis
+        Get VM objects by name from the VMWare Rest API.  Supports wildcards and handles case sensitivity issues.
+
+    .DESCRIPTION
+        Get details about vm or a list of vms from VMWare Rest API.  Supports wildcards and handles case sensitivity issues.
+        Able to return more than 1000 objects by working around the current limitation of the REST APIs.
+
+    .PARAMETER vCenter
+        FQDN of server to connect to
+
+    .PARAMETER sessionID
+        vmware-api-session-id from Connect-vmwwarerasession
+
+    .PARAMETER name
+        name of the VM, case sensitivity not required. Supports wildcard character *.
+
+    .PARAMETER host
+        name of host to return list of VMs from, case sensitivity required
+
+    .PARAMETER hostID
+        ID of host to return list of VMs from. Overrides host parameter if also specified.
+
+    .PARAMETER cluster
+        Name of cluster to return list of VMs from (able to return more then 1000 VMs), case sensitivity required.
+
+    .PARAMETER clusterID
+        ID of cluster to return list of VMs from (able to return more then 1000 VMs.) Overrides cluster parameter if specified.
+
+    .OUTPUTS
+        List of VM objects based on parameters specified.
+
+    .Notes
+        Author: Aaron Smith
+#>
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$vCenter,
+        
+        [Parameter(Mandatory)]
+        [string]$sessionID,
+
+        [string]$name,
+
+        [string]$host,
+
+        [string]$hostID,
+
+        [string]$cluster,
+
+        [string]$clusterID
+    )
+
+    Begin
+    {
+    }
+    Process
+    {
+        if ($cluster)
+        {
+            $clusterID = (Get-VMWareRACluster -vCenter $vCenter -sessionID $sessionID -name $cluster).cluster;
+
+            if (!$clusterID)
+            {
+                throw "Cluster ${cluster} not found by name";
+            }
+        }
+
+        if ($host)
```

Comment:
If some one provides -host and -hostid, hostid will be over-written.  You may not care.

Reply:
That's by design.  Called out in the .PARAMETER for hostID.  Chances are high you'd only use one or the other in my opinion.  Splitting the logic apart so both could be used is possible, but then we have to account for the possibility that -host and -hostID return the same thing and potentially duplicates the results.

Reply:
the results would actually override -hostID, not make a duplicate.  But you can either be fine with that and its the user's responsibility to use the function correctly or you could do a check and see if they match or adapt your code to to stack filters.  I don't have an opinion I just wanted to make sure you know what would happen.

Reply:
Since the function documents the behavior, I'm fine with it as-is.  We can always change the behavior later if need be.  But I'll think about it more.  I'd want it consistent with the rest of the module, so if the behavior of the module is to NOT have one parameter be able to override another, then I'll modify via a separate pull request.  At the moment I consider parameters like -host and -hostID to be mutually exclusive.  But I can see how it'd be potentially confusing to a consumer of the module.